registry/client/transport/http_reader.go

```diff
@@ -2,15 +2,31 @@ package transport
 
 import (
 	"bufio"
-	"bytes"
 	"errors"
 	"fmt"
 	"io"
-	"io/ioutil"
 	"net/http"
 	"os"
 )
 
+// HTTPError encapsulates an unsuccessful HTTP status code.
+type HTTPError struct {
+	Resp       *http.Response
+	bodyClosed bool
+}
+
+// Error renders an HTTPError to a string.
+func (h *HTTPError) Error() string {
+	// If the Error method gets called, the error propagated somewhere
+	// where it's not being handled natively as a HTTPError, so we need
+	// to take care of closing the body.
+	if !h.bodyClosed {
+		h.Resp.Body.Close()
```

Comment:
@aaronlehmann This is how it was before. Actually, there was only one request open at any time. If it was still open on the `Close`, it would have the body closed, as well.